src/AppInstallerSharedLib/Yaml.cpp

```diff
@@ -146,6 +195,38 @@ namespace AppInstaller::YAML
         m_scalar = std::move(value);
     }
 
+    void Node::SetScalar(std::string value, bool isQuoted)
+    {
+        this->SetScalar(value);
+
+        // For untagged scalar nodes, libyaml always assigns the generic string
+        // tag. Here we just try our best and assume that if the value is unquoted
+        // then is not necessarily a string.
+        // TODO: handle float and timestamps
+        if (!isQuoted && this->GetTagType() == TagType::Str)
+        {
+            // Integer
+            // 0 | -? [1-9] [0-9]*
+            try
+            {
+                this->as<int64_t>();

```

Comment:
This seems like it might be very noisy for logs.  Could we have a `try_as` that returns a `std::optional` and uses functions that don't end up logging?